ledger-service/http-json/src/main/scala/com/digitalasset/http/dbbackend/ContractDao.scala

```diff
@@ -36,10 +36,6 @@ object ContractDao {
     "oracle.jdbc.OracleDriver" -> SupportedJdbcDriver.Oracle,
   )
 
-  lazy val supportedJdbcDriverNames = supportedJdbcDrivers.keySet filter { d =>
-    scala.util.Try(Class forName d).isSuccess
-  }
```

Comment:
Instead, this should be `supportedJdbcDrivers.keySet intersect JdbcDrivers.supportedJdbcDriverNames` with the change in http-json `Main` reverted.

Reply:
Why do you prefer that option? I’d rather move down the opposite direction and have the switch between CE/EE as far up the dependency tree as possible.

Reply:
I'm not moving that switch with this suggestion, merely making it so that adding `foo` to `JdbcDrivers` doesn't imply a promise to immediately support `foo` in json-api (which `--help` will advertise).

Reply:
Makes sense, fixed in 1febf666ea9fc151ddc49558bf9b16fbc032650a. I’ve slightly modified it such that the final switch is still in `Main`. That way we have a chance of getting to the point where we don’t have to compile `ContractDao` twice.